### PR TITLE
Adds IncludePlatform and ExcludePlatform properties to the TestCaseAttribute

### LINKED ISSUE #20: TestCaseAttribute needs Platform property
There should be a way to add a TestCase conditioned on the platform.

From https://bugs.launchpad.net/nunit-3.0/+bug/608970

### PR DESCRIPTION
**Pull request by rprouse**
Fixes #20

### CODE REVIEW

File: src/NUnitFramework/framework/Attributes/TestCaseAttribute.cs [hunk 1]
```diff
@@ -384,7 +399,17 @@ private static void PerformSpecialConversions(object[] arglist, ParameterInfo[]
         /// <returns>One or more TestMethods</returns>
         public IEnumerable<TestMethod> BuildFrom(MethodInfo method, Test suite)
         {
-            return new TestMethod[] { new NUnitTestCaseBuilder().BuildTestMethod(method, suite, GetParametersForTestCase(method)) };
+            TestMethod test = new NUnitTestCaseBuilder().BuildTestMethod(method, suite, GetParametersForTestCase(method));
+            
+#if !PORTABLE
+            if (test.RunState != RunState.NotRunnable && !_platformHelper.IsPlatformSupported(this))
```
Comment (CharliePoole) [inline src/NUnitFramework/framework/Attributes/TestCaseAttribute.cs]: You should also exclude the case where RunState is already Ignored.

**Comment by CharliePoole** [comment]
Looks good to me if you make the one change I noted inline.

File: src/NUnitFramework/framework/Attributes/TestCaseAttribute.cs (hunk 1 above)
Comment (rprouse) [inline src/NUnitFramework/framework/Attributes/TestCaseAttribute.cs]: This code is copied from PlatformAttribute, should it be added there too?

Comment (CharliePoole) [inline src/NUnitFramework/framework/Attributes/TestCaseAttribute.cs]: I guess so... I just took a look at it and if you have both a PlatformAttribute and an IgnoreAttribute, the resulting RunState is indeterminate.

FWIW, this is a carryover from NUnit 2.x, when all attributes were applied in a known order. Now that they apply themselves, each one has to allow for the fact that other attributes may be present.

I'd say fix this instance and we can do a more systematic review later, possibly under the aegis of #374.

**Comment by rprouse** [comment]
I've made the change you requested to TestCaseAttribute and PlatformAttribute. You or I can merge once Travis and AppVeyor are happy.

**Comment by CharliePoole** [comment]
It builds under CF.

**Pull request merged**